test/unit/alignment/band/static_band_test.cpp

```diff
@@ -0,0 +1,34 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2018, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include <gtest/gtest.h>
+
+#include <seqan3/alignment/band/static_band.hpp>
+
+using namespace seqan3;
+
+TEST(static_band, static_band)
+{
+    static_band bs{lower_bound{-2}, upper_bound{2}};
+    EXPECT_EQ(bs.lower_bound, -2);
+    EXPECT_EQ(bs.upper_bound, 2);
+    EXPECT_TRUE((std::is_same_v<decltype(bs), static_band<int>>));
+}
+
+TEST(static_band, various_integer_types)
+{
+    static_band b{lower_bound{static_cast<int8_t>(-5)}, upper_bound{static_cast<int8_t>(5)}};
+    EXPECT_TRUE((std::is_same_v<decltype(b), static_band<int8_t>>));
+
+    static_band b2{lower_bound{static_cast<int16_t>(-5)}, upper_bound{static_cast<int16_t>(5)}};
+    EXPECT_TRUE((std::is_same_v<decltype(b2), static_band<int16_t>>));
```

Comment:
Is the type of these parameters down-the-line dependent on the input type? Why not always use uint64_t to reduce template instantiations? I guess you use different internal types for the kernel anyway?